Zoom Buttons/ZoomButtons.js

```diff
@@ -0,0 +1,201 @@
+tau
+    .mashups
+    .addDependency('jQuery')
+    .addDependency('tau/configurator')
+    .addMashup(function($, configurator) {
+
+        'use strict';
+
+        var zoomTimer = null;
+
+        var reg = configurator.getBusRegistry();
+
+        var addBusListener = function(busName, eventName, listener) {
+
+            reg.on('create', function(e, data) {
+
+                var bus = data.bus;
+                if (bus.name === busName) {
+                    bus.on(eventName, listener);
+
+                }
+            });
+
+            reg.on('destroy', function(e, data) {
+
+                var bus = data.bus;
+                if (bus.name === busName) {
+                    bus.removeListener(eventName, listener);
+                }
+
+                if ( zoomTimer !== null ) {
+                    clearTimeout( zoomTimer );
+                }
+
+            });
+
+            reg.getByName(busName).done(function(bus) {
+                bus.on(eventName, listener);
+            });
+        };
+
+        var appendZoomButtons = function appendFullscreenButton(e, renderData) {
+            var $el = renderData.element;
+
+            if (!$el.find('.btnZoom').length) {
+                var $elWrapper = $el.find('.tau-board-header__control--actions');
+                var $elAnchor = $elWrapper.length ? $elWrapper: $el.find('[role=actions-button]').parent();
+
+                var $elGroup = $('<div id="zoom-buttons" class="tau-board-header__control--mashup" style="display:flex;margin-left:10px;align-items:center;"></div>');
+
+                $.each({
+                    Min: 1,
+                    XS:  2,
+                    S:   3,
+                    M:   4,
+                    L:   5,
+                    XL:  6
+                }, function( name, zoom ) {
+                    $elGroup.append( $(
+                        '<button class="tau-btn tau-btn--icon btnZoom" id="btnZoom'+ zoom +'" data-zoom="'+ zoom +'" ><span class="tau-btn__icon">'+ name +'</span></button>'
+                    ) );
+                });
+
+                $elAnchor.before( $elGroup  );
+            }
+
+            // initially we have to open the menu and wait until the slider is loaded in dom, or reading slider does not work
+            // try x times and then give up, maybe its dashboard and we dont want to poll
+
+            // initialize when buttons are rendered, which is also boardchange
+            waitForSlider(10); // try 10 times and then give up, might be dashboard
+
+        };
+
+        var toolbarComponents = [
+            'board.toolbar'
+        ];
+
+        toolbarComponents.forEach(function(componentName) {
+            addBusListener(componentName, 'afterRender', appendZoomButtons);
+        });
+
+        $(document).on('click','.btnZoom', function(){
+            var zoom = $(this).attr('data-zoom');
+
+            var was_open = false;
```

Comment:
Prefer to assign `isActionMenuOpen()` result directly here and below, like this:
```
var was_open = isActionMenuOpen();
```

Also, need to fix `isActionMenuOpen` to return `Boolean` (notice `is Something` notation):

```
var isActionMenuOpen = function() {
     return $('.tau-board-header__control--actions .tau-actions-btn.tau-checked').length !== 0;
};
```

Reply:
use the version which is better, i will use it in future. 
i wrote the mashup just with jquery and it worked. maybe there are more efficient ways to do it, i don't know enough how to use API functions directly. 
Basically this only remote-controlls the hidden zoom slider and makes the buttons available again in the toolbar. i change the card-size all the time, and its really annoying it was moved into the action menu. in old TP Videos you can see the original buttons. so i recreated this great feature.

Reply:
Well, it is our fault. 

There are not so many API's exposed externally (https://dev.targetprocess.com/), so it is hard to write reliable mashups when you do not know internal TP object structures. We'll slowly add new APIs and expose them, but it takes some time.